app/models/evaluation_rubric.php

```diff
@@ -393,10 +393,22 @@ function rubricEvalScore($eventId, $conditions) {
         $sub = $evalSub->getEvalSubmissionsByEventId($eventId);
         $event = $this->Event->find('first', array('conditions' => array('Event.id' => $eventId)));
         $submitted = Set::extract('/EvaluationSubmission/submitter_id', $sub);
-
-        $rubricDetails = $this->find('all', array(
-            'conditions' => array_merge(array('evaluator' => $submitted), $conditions)
-        ));
+        
+        
+        $rubricDetails = array();
+        
+        //chunk find
+        $total = $this->find('count', array('conditions' => array_merge(array('evaluator' => $submitted), $conditions) ));
+        $limit = 100;
+        $pages = ceil($total / $limit);
+        for ($page = 1; $page < $pages + 1; $page++) {
```

Comment:
It uses less memory overall. With larger classes this query is fetching 1500+ records which was the main contributing factor to timeouts. Fetching the results in chunks greatly reduced the amount of memory being used (at least in the development environment). It might have to do with how cakephp performs the find function but I didn't investigate it further.